raiden/network/transport/matrix/transport.py

```diff
@@ -644,6 +644,7 @@ def _handle_invite(self, room_id: _RoomID, state: dict):
         last_ex = None
         for _ in range(JOIN_RETRIES):
             try:
+                gevent.sleep(.1)
```

Comment:
Maybe using exponential backoff here would make sense. We already have utilities for that.

Reply:
@ulope 
This is a little odd; i implemented the exponential back off, but the tests still required `JOIN_RETRIES >= 70` to pass. 
The tests pass (at least locally) with `greenlet.sleep()`, as long as the for loop has at least some 70 iterations. I think the cause is not timeouts, but that the `_handle_invite` greenlet blocks a room creation greenlet from being processed in the tests.

Reply:
There was a race in the tests where the room creation was already successful, but another node requested the room before it was propagated. This was solved now with another retry.